### PR TITLE
explorer: adding support for multi-line tab and SHIFT + Tab

### PR DESCRIPTION
**Pull request by mike-hunhoff**
addresses #471.

https://user-images.githubusercontent.com/42192796/110872888-e04bdd80-828d-11eb-861b-e89fbf64679f.mov

### CODE REVIEW

File: capa/ida/plugin/view.py [hunk 1]
```diff
@@ -210,12 +213,99 @@ def load_preview_meta(self, ea, author, scope):
...
-            self.insertPlainText(" " * 2)
+        """intercept key press events"""
+        if e.key() in (QtCore.Qt.Key_Tab, QtCore.Qt.Key_Backtab):
+            # apparently it's not easy to implement tabs as spaces, or multi-line tab or SHIFT + Tab
+            # so we need to implement it ourselves so we can retain properly formatted capa rules
+            # when a user uses the Tab key
+            if self.textCursor().selection().isEmpty():
+                # single line, only worry about Tab
+                if e.key() == QtCore.Qt.Key_Tab:
+                    self.insertPlainText(self.INDENT)
```
Comment (Ana06) [inline capa/ida/plugin/view.py:224]: Is this needed? When nothing is selected tab already work 🤔

**Review by Ana06 — COMMENTED** [review]
This is great! 🎉 Thank you @mike-hunhoff!

File: capa/ida/plugin/view.py [hunk 2]
```diff
@@ -178,6 +178,9 @@ def build_context_menu(o, actions):
 
 
 class CapaExplorerRulgenPreview(QtWidgets.QTextEdit):
+
+    INDENT = " " * 2
```
Comment (mr-tz) [inline capa/ida/plugin/view.py:182]: space indent 🤟

File: capa/ida/plugin/view.py (hunk 1 above)
Comment (mr-tz) [reply capa/ida/plugin/view.py:224]: It may be needed because this handles all `tab` key presses.

File: capa/ida/plugin/view.py [hunk 3]
```diff
@@ -178,6 +178,9 @@ def build_context_menu(o, actions):
 
 
 class CapaExplorerRulgenPreview(QtWidgets.QTextEdit):
```
Comment (mr-tz) [inline capa/ida/plugin/view.py:180]: missing an e in "Rule", but no biggie and not related here at all

**Review by mr-tz — APPROVED** [review]
nice!

on a different not: some of the plugin files are getting quite large, maybe we can refactor/split those up?

File: capa/ida/plugin/view.py (hunk 3 above)
Comment (williballenthin) [reply capa/ida/plugin/view.py:180]:
```suggestion
class CapaExplorerRulegenPreview(QtWidgets.QTextEdit):
```

references will need fixup too, though

**Review by williballenthin — APPROVED** [review]

**Review by Ana06 — APPROVED** [review]

File: capa/ida/plugin/view.py (hunk 3 above)
Comment (mike-hunhoff) [reply capa/ida/plugin/view.py:180]: let's save this for #488.

**Pull request merged**